api/graphql/types/recurring_reservation/serializers.py

```diff
@@ -38,52 +47,366 @@ class Meta:
+            )
+            for begin, end in periods:
+                if begin.date() in skip:
+                    continue
+
+                timespan = TimeSpanElement(
+                    start_datetime=begin,
+                    end_datetime=end,
+                    is_reservable=True,
+                    buffer_time_after=instance.reservation_unit.buffer_time_after,
+                    buffer_time_before=instance.reservation_unit.buffer_time_before,
+                )
+
+                # Would the reservation overlap with any existing reservations?
+                # Also checks that buffers for the reservation will also not overlap.
+                if any(timespan.overlaps_with(reserved) for reserved in reserved_timespans):
+                    overlapping.append(
+                        ReservationPeriodJSON(
+                            begin=begin.isoformat(timespec="seconds"),
+                            end=end.isoformat(timespec="seconds"),
+                        )
+                    )
+                    continue
+
+                # Would the reservation be fully inside any reservable timespans for the resource?
+                # Ignores buffers for the reservation, since those can be outside reservable times.
+                if check_opening_hours and not any(
+                    timespan.fully_inside_of(reservable) for reservable in reservable_timespans
+                ):
+                    not_reservable.append(
+                        ReservationPeriodJSON(
+                            begin=begin.isoformat(timespec="seconds"),
+                            end=end.isoformat(timespec="seconds"),
+                        )
+                    )
+                    continue
+
+                non_overlapping.append(ReservationPeriod(begin=begin, end=end))
+
+        if overlapping:
+            msg = "Not all reservations can be made due to overlapping reservations."
+            extensions = {"code": error_codes.RESERVATION_SERIES_OVERLAPS, "overlapping": overlapping}
+            raise GraphQLError(msg, extensions=extensions)
+
+        if not_reservable:
+            msg = "Not all reservations can be made due to falling outside reservable times."
+            extensions = {"code": error_codes.RESERVATION_SERIES_NOT_OPEN, "not_reservable": not_reservable}
+            raise GraphQLError(msg, extensions=extensions)
+
+        # Pick out the through model for the many-to-many relationship and use if for bulk creation
+        ThroughModel: type[models.Model] = Reservation.reservation_unit.through  # noqa: N806
+
+        reservations: list[Reservation] = []
+        through_models: list[models.Model] = []
+
+        for period in non_overlapping:
+            reservation = Reservation(
+                begin=period["begin"],
+                end=period["end"],
+                recurring_reservation=instance,
+                age_group=instance.age_group,
+                **reservation_details,
```

Comment:
`buffer_before/after` needs to account for `reservation_unit.reservation_block_whole_day` (on basis of the date of each reservation): cf. https://github.com/City-of-Helsinki/tilavarauspalvelu-core/blob/main/api/graphql/types/reservation/serializers/staff_create_serializers.py#L150C16-L150C61

Reply:
Added support for this.